app/components/clusters/cluster-expand.ts

```diff
@@ -0,0 +1,311 @@
+// <reference path="../typings/tsd.d.ts" />
+
+import {OSDService} from '../rest/osd';
+import {PoolService} from '../rest/pool';
+import {UtilService} from '../rest/util';
+import {VolumeService} from '../rest/volume';
+import {ServerService} from '../rest/server';
+import {ClusterService} from '../rest/clusters';
+import {RequestService} from '../rest/request';
+import {ClusterHelper} from './cluster-helpers';
+import {RequestTrackingService} from '../requests/request-tracking-svc';
+import * as ModalHelpers from '../modal/modal-helpers';
+
+export class ClusterExpandController {
+    private name: any;
+    private cluster: any;
+    private newHost: any;
+    private clusterID: number;
+    private hosts: Array<any>;
+    private disks: Array<any>;
+    private osds: Array<any>;
+    private clusterHelper: ClusterHelper;
+    
+    static $inject: Array<string> = [
+        '$q',
+        '$log',
+        '$scope',
+        '$modal',
+        '$timeout',
+        '$location',
+        '$routeParams',
+        'OSDService',
+        'UtilService',
+        'ServerService',
+        'ClusterService',
+        'RequestService',
+        'RequestTrackingService'
+    ];
+
+    constructor(private qService: ng.IQService,
+        private logService: ng.ILogService,
+        private scopeService: ng.IScope,
+        private modalService: any,
+        private timeoutService: ng.ITimeoutService,
+        private locationService: ng.ILocationService,
+        private routeParamsSvc: ng.route.IRouteParamsService,
+        private osdService: OSDService,
+        private utilService: UtilService,
+        private serverService: ServerService,
+        private clusterService: ClusterService,
+        private requestService: RequestService,
+        private reqTrackingService: RequestTrackingService) {
+        
+        this.hosts = [];
+        this.disks = [];
+        this.osds = [];
+        this.clusterID =  this.routeParamsSvc['id'];
+        
+        this.clusterService.get(this.clusterID).then(this.clusterGetCallBack);
+        this.serverService.getDiscoveredHosts().then(this.discoveredHostCallBack);
+        this.serverService.getFreeHosts().then(this.freeHostCallBack);
+    }
+
+    public clusterGetCallBack = (cluster: any) => {
```

Comment:
rename it to loadCluster. Change this to a normal function and use arrow function above